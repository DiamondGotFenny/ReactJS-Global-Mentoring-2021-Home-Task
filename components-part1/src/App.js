import './App.css';
import Home from './pages/Home';
import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            exact
            path="/"
            element={<Navigate to="/search" replace={true} />}
          />
          <Route path="/search" element={<Home />} />
          <Route path="/search/:searchQuery" element={<Home />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
