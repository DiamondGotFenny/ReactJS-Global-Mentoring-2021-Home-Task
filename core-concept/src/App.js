import React, { useState, useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Button from 'react-bootstrap/Button';
import uniqueId from 'lodash/uniqueId';
import './App.css';
import TodoItems from './Components/TodoItems';

const appHeader = React.createElement('h1', null, 'My Todo List');

function App() {
  const ITEMS_LIST = 'itemsList';
  const itemsListIntial = JSON.parse(localStorage.getItem(ITEMS_LIST)) || [];
  const [itemsList, setItemsList] = useState(itemsListIntial);

  const [inputVal, setInputVal] = useState('');

  const handleAddItem = () => {
    if (inputVal === '') {
      return;
    }

    const newItem = {
      id: uniqueId() + Date.now(),
      value: inputVal,
      isComplete: false,
      dateTime: new Date().toLocaleString(),
    };

    setItemsList([...itemsList, newItem]);
    setInputVal('');
  };

  const handleDeleteItem = (id) => {
    const newItemsList = itemsList.filter((item) => item.id !== id);
    setItemsList(newItemsList);
  };

  const handleCompleteItem = (id) => {
    const newItemsList = itemsList.map((item) => {
      if (item.id === id) {
        item.isComplete = !item.isComplete;
      }
      return item;
    });
    setItemsList(newItemsList);
  };

  useEffect(() => {
    localStorage.setItem(ITEMS_LIST, JSON.stringify(itemsList));
  }, [itemsList]);
  console.log(itemsList, ITEMS_LIST);
  return (
    <div className="App">
      {appHeader}
      <div className="container">
        <InputGroup className="inputGroup">
          <Button id="addItemBtn" onClick={handleAddItem}>
            +
          </Button>
          <FormControl
            type="text"
            placeholder="Enter Your To-Do Item Here"
            aria-label="Input to-do item"
            aria-describedby="addItemBtn"
            value={inputVal}
            onChange={(e) => setInputVal(e.target.value)}
          />
        </InputGroup>
        <TodoItems
          itemsList={itemsList}
          onCheck={handleCompleteItem}
          onDelete={handleDeleteItem}
        />
      </div>
    </div>
  );
}

export default App;
