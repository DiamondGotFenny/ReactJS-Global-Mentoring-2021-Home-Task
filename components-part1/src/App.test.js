import { render, screen } from '@testing-library/react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

//we need to overwrite the useSelector function to writable
//so that we can return a mock location object
Object.defineProperty(reactRedux, 'useSelector', {
  value: jest.fn(),
  configurable: true,
  writable: true,
});

const setup = (props = {}) => {
  const component = shallow(<App {...props} />);
  return component;
};

test('renders App without crash', () => {
  const mockDeletedMovie = '';
  const useLocationSpy = jest
    .spyOn(reactRedux, 'useSelector')
    .mockReturnValue(mockDeletedMovie);
  render(<App />);
  expect(useLocationSpy).toHaveBeenCalled();
  const contentEle = screen.getByText(/found/i);
  expect(contentEle).toBeInTheDocument();
});

it('renders without crashing', () => {
  setup();
});

it('renders correctly', () => {
  const tree = toJson(setup());
  expect(tree).toMatchSnapshot();
});
