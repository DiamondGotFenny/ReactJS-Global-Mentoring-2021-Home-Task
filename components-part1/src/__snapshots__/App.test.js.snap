// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Routes>
      <Route
        element={
          <Navigate
            replace={true}
            to="/search"
          />
        }
        exact={true}
        path="/"
      />
      <Route
        element={<Memo(Home) />}
        path="/search"
      />
      <Route
        element={<Memo(Home) />}
        path="/search/:searchQuery"
      />
    </Routes>
  </BrowserRouter>
</div>
`;
